File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.TrainingService.Models/iget-training-events_item.ts



import { GetTrainingEventAppUser_Item } from './get-training-event-app-user_item';
import { GetTrainingEventLocation_Item } from './get-training-event-location_item';
import { GetTrainingEventUSPartnerAgency_Item } from './get-training-event-uspartner-agency_item';
import { GetTrainingEventProjectCode_Item } from './get-training-event-project-code_item';
import { GetTrainingEventStakeholder_Item } from './get-training-event-stakeholder_item';
import { GetTrainingEventAttachment_Item } from './get-training-event-attachment_item';

export interface IGetTrainingEvents_Item {
  
	TrainingEventID: number;
	Name: string;
	NameInLocalLang: string;
	TrainingEventTypeID: number;
	TrainingEventTypeName: string;
	Justification: string;
	Objectives: string;
	ParticipantProfile: string;
	SpecialRequirements: string;
	ProgramID?: number;
	KeyActivityID?: number;
	KeyActivityName: string;
	TrainingUnitID?: number;
	BusinessUnitAcronym: string;
	BusinessUnitName: string;
	CountryID?: number;
	PostID?: number;
	ConsularDistrictID?: number;
	OrganizerAppUserID?: number;
	PlannedParticipantCnt?: number;
	PlannedMissionDirectHireCnt?: number;
	PlannedNonMissionDirectHireCnt?: number;
	PlannedMissionOutsourceCnt?: number;
	PlannedOtherCnt?: number;
	EstimatedBudget?: number;
	ActualBudget?: number;
	EstimatedStudents?: number;
	FundingSourceID?: number;
	AuthorizingLawID?: number;
	ModifiedByAppUserID?: number;
	ModifiedDate?: Date;
	TravelStartDate?: Date;
	TravelEndDate?: Date;
	EventStartDate?: Date;
	EventEndDate?: Date;
	StudentCount?: number;
	InstructorCount?: number;
	TrainingEventStatusID?: number;
	TrainingEventStatus: string;
	Organizer?: GetTrainingEventAppUser_Item;
	ModifiedBy?: GetTrainingEventAppUser_Item;
	TrainingEventLocations?: GetTrainingEventLocation_Item[];
	TrainingEventUSPartnerAgencies?: GetTrainingEventUSPartnerAgency_Item[];
	TrainingEventProjectCodes?: GetTrainingEventProjectCode_Item[];
	TrainingEventStakeholders?: GetTrainingEventStakeholder_Item[];
	TrainingEventAttachments?: GetTrainingEventAttachment_Item[];

}

File: GTTS/INL.TrainingService.Models/IGetTrainingEvents_Item.cs
using System;
using System.Collections.Generic;

namespace INL.TrainingService.Models
{
    interface IGetTrainingEvents_Item
    {
        long TrainingEventID { get; set; }
        string Name { get; set; }
        string NameInLocalLang { get; set; }
        string TrainingEventType { get; set; }
        int? ProgramID { get; set; }
        string BusinessUnitAcronym { get; set; }
        string BusinessUnit { get; set; }
        int? OrganizerAppUserID { get; set; }
        string Organizer { get; set; }
        int? TrainingEventStatusID { get; set; }
        string TrainingEventStatus { get; set; }
        DateTime? EventStartDate { get; set; }
        DateTime? EventEndDate { get; set; }
        int? ModifiedByAppUserID { get; set; }
        string ModifiedByAppUser { get; set; }
        DateTime? ModifiedDate { get; set; }
        DateTime? CreatedDate { get; set; }  

        List<GetTrainingEventLocation_Item> TrainingEventLocations { get; set; }
        List<GetTrainingEventKeyActivity_Item> TrainingEventKeyActivities { get; set; }
    }
}
